python/ray/serve/_private/application_state.py

```diff
@@ -81,18 +82,28 @@ def deploy(self, deployment_params: List[Dict]) -> List[str]:
 
         # Update route prefix for application
         num_route_prefixes = 0
+        num_fastapi_deployments = 0
         for deploy_param in deployment_params:
             if (
                 "route_prefix" in deploy_param
                 and deploy_param["route_prefix"] is not None
             ):
                 self.route_prefix = deploy_param["route_prefix"]
                 num_route_prefixes += 1
+
+            if "docs_path" in deploy_param and deploy_param["docs_path"] is not None:
+                self.docs_path = deploy_param["docs_path"]
+                num_fastapi_deployments += 1
         assert num_route_prefixes <= 1, (
             f"Found multiple route prefix from application {self.name},"
             " Please specify only one route prefix for the application "
             "to avoid this issue."
         )
+        assert num_fastapi_deployments <= 1, (
+            f"Found multiple FastAPI deployments from application {self.name}. "
+            "Please only use one FastAPI deployment in your application to avoid "
+            "this issue."
+        )
```

Comment:
please add test cases for these! we should also not use `assert` for any user-facing error messages (it's written like it might be one)

Reply:
Added! I also changed the error message to be more generic in case we decide to expand usage of `docs_path` in the future.

Reply:
nice thank you